### ISSUE TITLE
[FEATURE] Disallow manual bankruptcy

### ISSUE THREAD

**Issue by chungfg** [COMP3122-2324-Project/Simpsyber#26]
Currently, if a player does not have enough money but continue doing environment (QA/Deployment) upgrade, it will cause a bankruptcy. I think the game should block such actions. Since the player does not have enough budget to do the upgrade, the action should not establish in the game and this action also does not make sense in reality.

https://github.com/COMP3122-2324-Project/Simpsyber/assets/82433114/7062aee9-4563-4a46-b0a9-c690094acc24

**Comment by Samfung0612** [comment]
I completely agree with your suggestion. The button should be disabled if there isn't enough budget available for an upgrade. To prevent this situation from occurring again, I propose adding a notification on the upgrade page. This notification would clearly state when there is insufficient budget to upgrade the environment.
the flowing image show the possible ways to show that not enough budget to upgrade.
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/69749799/13a3c8b0-ff72-4384-b409-5a344cc1c851)
